packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m

```diff
@@ -83,26 +81,38 @@ - (NSDictionary *)constantsToExport {
     :(RCTPromiseResolveBlock) resolve
     :(RCTPromiseRejectBlock) reject
 ) {
+  #if !(TARGET_IPHONE_SIMULATOR)
   if ([UIApplication sharedApplication].isRegisteredForRemoteNotifications == NO) {
     [RNFBSharedUtils rejectPromiseWithUserInfo:reject userInfo:(NSMutableDictionary *) @{
         @"code": @"unregistered",
-        @"message": @"You must be registered for remote notifications before calling get token, see messaging().registerForRemoteNotifications() or requestPermission().",
+        @"message": @"You must be registered for remote messages before calling getToken, see messaging().registerDeviceForRemoteMessages().",
     }];
     return;
   }
+  #endif
 
-  NSDictionary *options = nil;
-  if ([FIRMessaging messaging].APNSToken) {
-    options = @{@"apns_token": [FIRMessaging messaging].APNSToken};
-  }
-
-  [[FIRInstanceID instanceID] tokenWithAuthorizedEntity:authorizedEntity scope:scope options:options handler:^(NSString *_Nullable identity, NSError *_Nullable error) {
-    if (error) {
-      [RNFBSharedUtils rejectPromiseWithNSError:reject error:error];
-    } else {
-      resolve(identity);
+  if ([scope isEqualToString:@"FCM"] && [authorizedEntity isEqualToString:[FIRApp defaultApp].options.GCMSenderID]) {
+    [[FIRInstanceID instanceID] instanceIDWithHandler:^(FIRInstanceIDResult *_Nullable result, NSError *_Nullable error) {
+      if (error) {
+        [RNFBSharedUtils rejectPromiseWithNSError:reject error:error];
+      } else {
+        resolve(result.token);
+      }
+    }];
+  } else {
+    NSDictionary *options = nil;
+    if ([FIRMessaging messaging].APNSToken) {
```

Comment:
If the APNSToken is not set (returning user), an 1001 error is likely to be thrown. Showing a warning message "did you forget to call registerForRemoteNotifications() ?" could help.